packages/block-library/src/query/edit/query-content.js

```diff
@@ -106,21 +98,29 @@ export default function QueryContent( {
 		} else if ( ! query.perPage && postsPerPage ) {
 			newQuery.perPage = postsPerPage;
 		}
-		// We need to reset the `inherit` value if not in a template, as queries
-		// are not inherited when outside a template (e.g. when in singular content).
-		if ( ! isTemplate && query.inherit ) {
+		// We need to reset the `inherit` value if in a singular template, as queries
+		// are not inherited when in singular content (e.g. post, page, 404, blank).
+		if ( isSingular && query.inherit ) {
 			newQuery.inherit = false;
 		}
+		// We need to update the query in the Editor if a specific post type is set.
+		// Unless the post type is `page`, as it usually doesn't make sense to loop
+		// through pages.
+		if ( postType && postType !== 'page' && query.postType !== postType ) {
+			newQuery.postType = postType;
```

Comment:
Why not set use this context straight in the `post-template` block instead of editing the block on mount by setting attributes?

Reply:
I like the sound of this, but when I try moving the `postType` context to the `post-template` block, the Query Loop attributes aren't updated and so the incorrect post type is used in the inspector controls on mount, even though the correct post type is used in the query preview itself. For example, if I create a template for a custom post type and insert a Query Loop block, I'd expect the custom post type to be used, but instead it's always defaulting to `post`, even though the Query Loop block itself is displaying posts from the correct post type:

<img width="282" alt="image" src="https://github.com/user-attachments/assets/a13291c5-df24-4d23-b81a-9def2055e9eb">

It looks like the attributes need to be updated in order for these controls to stay up to date (or we keep the context near the controls in `query-content`).

Reply:
Yes, checking the context in the controls seems to work! I've tried that out here: https://github.com/WordPress/gutenberg/pull/65820/commits/aec7cba4f37baa3707e6cf7deebf7bd690cd9ce1.

Rather than disabling the control, we want to read the post type from the context so that `post` is not always used by default when inside a custom post type.